master/internal/user/authz_rbac.go

```diff
@@ -198,6 +198,91 @@ func (a *UserAuthZRBAC) CanSetUsersRemote(ctx context.Context, curUser model.Use
 	return canAdministrateUser(ctx, curUser.ID)
 }
 
+// CanCreateUsersOwnToken always returns nil.
+func (a *UserAuthZRBAC) CanCreateUsersOwnToken(ctx context.Context, curUser model.User) error {
+	noPermissionRequired(ctx, curUser.ID, curUser.ID)
```

Comment:
This will have to double checked with product, in the event we don't want to allow users with the `Viewer` role to have access to this feature

Reply:
Resolving this comment as we have a consensus on this ask and have pushed those changes.